project1/src/main/java/uk/co/cpsd/javaproject1/SimulatorFrame.java

```diff
@@ -24,16 +26,24 @@ public SimulatorFrame() {
 
         // Set up simulation timer (1 tick per 500ms)
         timer = new Timer(1000, (ActionEvent e) -> {
-            world.tick();
-            worldPanel.repaint();
+            world.moveAnimals();        // Make goats move + eat + act
+            world.tick();               // Tick: grow grass, energy, reproduction
+            worldPanel.repaint();       // Show updates on the screen
+        
+            if (world.getSecondsElapsed() >= tickLimitSimulation) {
+                world.writeToCSV(world.getGoatHistory(), world.getGrassHistory());
+                timer.stop();
+        
+                SwingUtilities.invokeLater(() -> {
+                    GoatChart chart = new GoatChart(world.getGoatHistory(),world.getGrassHistory());
+                    chart.setVisible(true);
+                });
+            }
         });
 
-        goatTimer = new Timer(2000,(ActionEvent e)->{
-            world.moveAnimals();
-            // worldPanel.repaint();
-        });
         // Start the simulation
         timer.start();
-        goatTimer.start();
+        // goatTimer.start();
```

Comment:
Again: good code stays, bad code goes.  If you need to look back, you've got git history.